test/junit/scala/tools/nsc/reporters/AbstractCodeActionTest.scala

```diff
@@ -91,9 +92,90 @@ abstract class AbstractCodeActionTest extends BytecodeTesting {
     )
   }
 
-  def assertCodeSuggestion(original: String, expected: String): Unit = {
-    val run = compiler.newRun()
-    run.compileSources(compiler.global.newSourceFile(original) :: Nil)
+  @Test
+  def unitLiteralType(): Unit = {
+    assertCodeSuggestion("class C { def f: () = () }", "class C { def f: Unit = () }")
+    assertCodeSuggestion("class C { def f(a: Any) = a match { case _: () => 0 } }", "class C { def f(a: Any) = a match { case _: Unit => 0 } }")
+  }
+
+  @Test
+  def constructorProcedureSyntax(): Unit = {
```

Comment:
Just to note that linting that the unit-valued method has parens for a param list is turned off for tests.

That makes sense as a policy because test methods are never called explicitly; though it would be even finer if only test-annotated methods were ignored.